### PR TITLE
Add support to specify the interpolation type for label images

### LINKED ISSUE #782: Support different default interpolation for LabelMaps
**🚀 Feature**
Allow to change default interpolation for labels

**Motivation**

When using augmentations, labels are interpolated using SITK nearest. Nearest seems to be producing some strange results, with interpolated labels being smaller than expected (I found this when comparing the same transforms applied with Monai vs torchio). I am attaching some images below as examples:

Original resolution
https://drive.google.com/file/d/1LKkT0vUui31eSGwJXiaVd_wsZxAwkmG1/view?usp=sharing

Slicing all dimensions [::2]
https://drive.google.com/file/d/1QbpjvUGF2PMqFa76hhdXKwUJQ262IyU9/view?usp=sharing

nearest (default)
https://drive.google.com/file/d/12Kk7sKs43X2j1DrIlauLSy0Fn9f8LTG6/view?usp=sharing

label-gaussian
https://drive.google.com/file/d/1cjje6pafVEqIk_xyHBzCdfvV5v1ulFum/view?usp=sharing

label_gaussian seems to be producing almost the same results as slicing and 'nearest' in MONAI 

**Pitch**

It would be great if torchio would support some sort of global configuration that would allow switching the default interpolation for labels to label-gaussian or alternatively a second parameter in augmentation

**Alternatives**

- A global configuration that applies automatically to all spatial transforms / augmentations
- A second parameter in relevant augmentations (i.e. image-interpolation='linear', label-interpolation='label-gaussian')

**Additional context**

Happy to provide other examples if needed. Thanks!

### PR DESCRIPTION
**Pull request by snavalm**
Fixes #782.

**Description**
Adds label_interpolation parameter in spatial augmentations and transforms

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/main/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [ ] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [x] In-line docstrings updated
- [ ] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Review by romainVala — COMMENTED** [review]
Looks good to me !
thanks for making it 

I think you also need to change the Doc at least in Doc/source/transforms/transforms.rst (line 178)

File: torchio/transforms/augmentation/spatial/random_elastic_deformation.py [hunk 1]
```diff
@@ -193,6 +198,7 @@ class ElasticDeformation(SpatialTransform):
         control_points:
         max_displacement:
         image_interpolation: See :ref:`Interpolation`.
+        label_interpolation: See :ref: Interpolation`.
```
Comment (fepegar) [inline torchio/transforms/augmentation/spatial/random_elastic_deformation.py:201]: I think a backtick is missing here.

**Review by fepegar — CHANGES_REQUESTED** [review]

**Comment by fepegar** [comment]
@snavalm could you modify the docs as suggested by @romainVala?

**Comment by snavalm** [comment]
Hi, sorry for the delay with this, I am off this week, I will try to do
this early next week

**Review by fepegar — CHANGES_REQUESTED** [review]
Could you please address https://github.com/fepegar/torchio/pull/791#issuecomment-1018342175?

**Comment by snavalm** [comment]
> ** fepegar ** request

Apologies,this is my first PR, trying to figure how to modify the docs... any pointers? I did `make html` as suggested here [https://github.com/fepegar/torchio/blob/main/CONTRIBUTING.rst](url), what should I do next?

**Review by fepegar — APPROVED** [review]
Thanks! This looks good. I will maybe add some minor changes and merge.

**Comment by fepegar** [comment]
Many thanks for your contribution, @snavalm! May I add you to the contributors list?

**Comment by snavalm** [comment]
Yes, please and thanks!

**Comment by fepegar** [comment]
@all-contributors please add @snavalm for code

**Pull request merged**